src/components/organisms/AssetActions/Edit/FormEditMetadata.tsx

```diff
@@ -31,6 +36,30 @@ export default function FormEditMetadata({
     setFieldValue(field.name, e.target.value)
   }
 
+  function handleTimeoutCustomOption() {
+    const timeoutInputIndex = data.findIndex(
+      (element) => element.name === 'timeout'
+    )
+    if (
+      data[timeoutInputIndex].options.length < 6 &&
+      !checkIfTimeoutInPredefinedValues(values.timeout)
+    ) {
+      data[timeoutInputIndex].options.push(values.timeout)
+    } else if (
+      data[timeoutInputIndex].options.length === 6 &&
+      checkIfTimeoutInPredefinedValues(values.timeout)
+    ) {
+      data[timeoutInputIndex].options.pop()
+    } else if (
+      data[timeoutInputIndex].options.length === 6 &&
+      data[timeoutInputIndex].options[5] !== values.timeout
+    ) {
+      data[timeoutInputIndex].options[5] = values.timeout
+    }
+  }
+
+  handleTimeoutCustomOption()
```

Comment:
this `handleTimeoutCustomOption()` shouldn't be directly in the component, use an effect

Reply:
`useEffect()` is not working too well. The option doesn't appear right away, I need to click the select then click elsewhere to make it render my option.

Reply:
problem is that in current implementation that method runs on every user keystroke so we need to prevent that somehow. Usually helps just testing a bunch of things in the `useEffect` dependency array to figure out upon which value change this should run. But yeah, can be lots of edge cases where UI does not update, pretty sure the whole Formic stuff adds to problems here too. In that case using some Formik method to update its values sometime helped

Reply:
it should work ok with `useEffect(()=>{handleTimeoutCustomOption()},[data])`

Reply:
I tried that but it doesn't, is the same problem that I described above

Reply:
I've done some testing and it will never work in the effect in the current format. When you update the array of the data object you will never trigger an event for react so `data` will need to be moved into a state and change all the things after. Fortunately because this component is handled by formik it's not rendered like a "normal" react component, `handleTimeoutCustomOption` is called only once so i think we don't need to bother with this anymore

Reply:
great investigation! So let's leave it as is and be done with it